### PR TITLE
Isotpsniffer add llopts support

### PR DESCRIPTION
**Pull request by rumpelsepp**

### CODE REVIEW

File: isotpsniffer.c [hunk 1]
```diff
@@ -302,6 +315,17 @@ int main(int argc, char **argv)
 
 	setsockopt(t, SOL_CAN_ISOTP, CAN_ISOTP_OPTS, &opts, sizeof(opts));
 
+	if ((setsockopt(s, SOL_CAN_ISOTP, CAN_ISOTP_LL_OPTS, &llopts, sizeof(llopts))) < 0) {
+		perror("setsockopt");
+		close(s);
```
Comment (hartkopp) [inline isotpsniffer.c]: is close(s) needed before exit?
AFAIK exit closes all file descriptors

File: isotpsniffer.c [hunk 2]
```diff
@@ -302,6 +315,17 @@ int main(int argc, char **argv)
...
+		exit(1);
+	}
+	if ((setsockopt(t, SOL_CAN_ISOTP, CAN_ISOTP_LL_OPTS, &llopts, sizeof(llopts))) < 0) {
+		perror("setsockopt");
+		close(s);
```
Comment (hartkopp) [inline isotpsniffer.c]: dito

File: isotpsniffer.c (hunk 1 above)
Comment (rumpelsepp) [reply isotpsniffer.c]: It does, but I learned in other projects that it is "good style" to close them explicitely.

**Comment by rumpelsepp** [comment]
> The if statement checks for "printable" ASCII chars - not for ASCII chars that are used in hex numbers. You were probably thinking of isprint() ?!?

You are right, I committed  a faulty version, since I did some experimentations.

File: isotpsniffer.c (hunk 1 above)
Comment (hartkopp) [reply isotpsniffer.c]: But then both sockets (s & t) need to be taken care of, right?

Comment (marckleinebudde) [reply isotpsniffer.c]: If you want to have proper error handling, please add it in a separate patch. Please use goto-style error handling, if you have more than one resource.

Comment (rumpelsepp) [reply isotpsniffer.c]: yep. better to fix the error handling properly before merging. will update this PR.

**Pull request merged**